web/src/test/auto/dom/cases/attachment/pageContextAttachment.js

```diff
@@ -0,0 +1,757 @@
+        await Promise.resolve();
+
+        // Note:  anything with iframes (design or not) requires an extra timeout for the internal doc to load.
+
+        assert.sameMembers(attached.map((elem) => elem.id), ['input']);
+        assert.sameMembers(attacher.inputList.map((elem) => elem.id), ['input']);
+        attacher.shutdown();
+      });
+
+      it('detachment: input and textarea', async function () {
+        fixture.load("input-and-text.html");
+        const attacher = this.attacher;
+        let attached = [];
+
+        attacher.on('enabled', (elem) => attached.push(elem));
+        attacher.install(false);
+
+        assert.sameMembers(attached.map((elem) => elem.id), ['input', 'textarea']);
+        assert.sameMembers(attacher.inputList.map((elem) => elem.id), ['input', 'textarea']);
+
+        // Now the fun part.
+        let detached = [];
+        attacher.on('disabled', (elem) => detached.push(elem));
+
+        fixture.cleanup("input-and-text.html");
+
+        // This gives the mutation observers a moment to 'kick in' and is required for test success.
+        await Promise.resolve();
+
+        assert.sameMembers(detached.map((elem) => elem.id), ['input', 'textarea']);
+        assert.sameMembers(attacher.inputList.map((elem) => elem.id), []);
+
+        attacher.shutdown();
+      });
+
+      it('content-editable div', async function () {
+        const attacher = this.attacher;
+        let attached = [];
+
+        attacher.on('enabled', (elem) => attached.push(elem));
+        attacher.install(false);
+        assert.sameMembers(attached.map((elem) => elem.id), []);
+
+        fixture.load("simple-editable-div.html");
+
+        // This gives the mutation observers a moment to 'kick in' and is required for test success.
+        await Promise.resolve();
+
+        assert.sameMembers(attached.map((elem) => elem.id), ['editable']);
+        assert.sameMembers(attacher.inputList.map((elem) => elem.id), ['editable']);
+        attacher.shutdown();
+      });
+
+      it('simple iframe with input', async function () {
+        const attacher = this.attacher;
+        let attached = [];
+
+        attacher.on('enabled', (elem) => attached.push(elem));
+        attacher.install(false);
+
+        fixture.load("simple-iframe-with-input.html");
+
+        // Note:  iframes require additional time to resolve.
+        await promiseForIframeLoad(document.getElementById('iframe'));
+
+        // Our mutation observers delay slightly here to ensure that any doc-internal handlers
+        // have a chance to resolve before we attach.  Currently: 10ms.
+        await timedPromise(20);
+
+        assert.sameMembers(attached.map((elem) => elem.id), ['iframe-input']);
+        assert.sameMembers(attacher.inputList.map((elem) => elem.id), ['iframe-input']);
+        attacher.shutdown();
+      });
+
+      it('detachment: simple iframe with input', async function () {
```

Comment:
This, along with other iframe-detachment tests after it, will likely fail if the bug-fix section mentioned in a prior comment is reverted.